components-compose/src/main/java/uk/gov/hmrc/components/compose/molecule/input/TextInputView.kt

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.hmrc.components.compose.molecule.input
+
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Clear
+import androidx.compose.material.icons.filled.Info
+import androidx.compose.material3.Icon
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.saveable.rememberSaveable
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.semantics.contentDescription
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextAlign
+import uk.gov.hmrc.components.compose.ui.theme.HmrcTheme.typography
+import uk.gov.hmrc.components.compose.ui.theme.hmrc_icon_size_24
+import uk.gov.hmrc.components.compose.ui.theme.hmrc_spacing_16
+import uk.gov.hmrc.components.compose.ui.theme.textInputViewColors
+
+@Composable
+fun TextInputView(
+    modifier: Modifier = Modifier,
+    onInputValueChange: ((String) -> Unit)? = null,
+    inputFilter: ((String, String) -> String)? = null,
+    placeholderText: String? = null,
+    errorText: String? = null,
+    errorContentDescription: String? = null,
+    labelText: String? = null,
+    labelContentDescription: String? = null,
+    singleLine: Boolean = false,
+    characterCount: Int? = null,
+    prefix: @Composable (() -> Unit)? = null,
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default
+) {
+
+    var localValue: String by rememberSaveable { mutableStateOf("") }
+
+    var localError: String? by rememberSaveable { mutableStateOf(null) }
+
+    localError = errorText
+
+    val counterEnabled: Boolean = characterCount != null
+
+    fun errorText(): @Composable (() -> Unit)? {
+        val text: String? = errorText
+        return text?.let {
```

Comment:
This val seems a bit unneccessary.. can we just use the errorText directly in the return statement?
```suggestion
        return errorText?.let {
```